solutions/cses-1095.mdx

```diff
@@ -0,0 +1,59 @@
+---
+id: cses-1712
+source: CSES
+title: Exponentiation II
+author: Ryan Chou
+---
+
+## Explanation
+
+[Fermat's Little Theorem](/gold/modular) tells us that $a^{p - 1} \equiv 1 \pmod{p}$, so we can calculate $a^{b^c \pmod{1e9 + 7 - 1}} \pmod{1e9+7}$ with modular exponentiation.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(\log P)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+const long long MOD = 1e9 + 7;
+
+//BeginCodeSnip{Binary Exponentiation}
+ll binpow(ll x, ll n, ll m) {
```

Comment:
Replace it with something shorter, like this
```c++
long long fast_pow(long long a, long long b, long long _MOD = 1e9 + 7) { 
    long long res = 1; 
    for (a %= _MOD; b; b >>= 1) { 
        if (b & 1) 
            res = res * a % _MOD; 
        a = a * a % _MOD; 
    } 
    return res; 
}
```

Reply:
that's how they do it in the module, ill just leave it as is for consistency